com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceCreator.java

```diff
@@ -33,14 +34,17 @@ public void createService(CloudFoundryOperations client, CloudServiceExtended se
         LOGGER.debug(format("Service \"{0}\" has defined service offering alternatives \"{1}\" for default service offering \"{2}\"",
             service.getName(), service.getServiceAlternatives(), service.getLabel()));
         List<String> possibleServiceOfferings = computePossibleServiceOfferings(service);
-        List<String> existingServiceOfferings = client.getServiceOfferings().stream().map(
-            cloudServiceOfferingExtended -> cloudServiceOfferingExtended.getLabel()).collect(Collectors.toList());
-        List<String> validServiceOfferings = computeValidServiceOfferings(client, possibleServiceOfferings, existingServiceOfferings);
+        // List<String> existingServiceOfferings = client.getServiceOfferings().stream().map(
+        // cloudServiceOfferingExtended -> cloudServiceOfferingExtended.getLabel()).collect(Collectors.toList());
+        Map<String, List<CloudServicePlan>> existingServiceOfferings = client.getServiceOfferings().stream().collect(
+            Collectors.toMap(CloudServiceOffering::getName, CloudServiceOffering::getCloudServicePlans));
+        List<String> validServiceOfferings = computeValidServiceOfferings(client, possibleServiceOfferings, service.getPlan(),
+            existingServiceOfferings);
 
         if (CollectionUtils.isEmpty(validServiceOfferings)) {
             LOGGER.error(format(
-                "Service \"{0}\" could not be created because any of defined service offering(s) \"{1}\" does not match with existing service offerings \"{2}\"",
-                service.getName(), possibleServiceOfferings, existingServiceOfferings));
+                "Service \"{0}\" could not be created because non of the defined service offering(s) \"{1}\" does not match with existing service offerings \"{2}\" or do not provide service plan \"{3}\"",
```

Comment:
Here we have double negation - "non of the services... does not match". It would be better to change it to "Service \"{0}\" could not be created because none of the service offering(s) \"{1}\" match with existing service offerings \"{2}\" or provide service plan \"{3}\""

Reply:
Thanks. Done

Reply:
Done